backend/src/main/kotlin/org/loculus/backend/utils/EarliestReleaseDateFinder.kt

```diff
@@ -0,0 +1,56 @@
+package org.loculus.backend.utils
+
+import kotlinx.datetime.LocalDate
+import kotlinx.datetime.LocalDateTime
+import kotlinx.datetime.LocalTime
+import mu.KotlinLogging
+import org.loculus.backend.service.submission.RawProcessedData
+
+private val log = KotlinLogging.logger { }
+
+/**
+ * Calculate the earliest release date for rows of sequence entries given to it one by one.
+ * Assumes that rows are sorted: all accession entries are given in a block, and with ascending versions.
```

Comment:
you mean add asserts in the class? I think it should be fine without adding stuff here, because I specifically added the test for that reason. But I'm also not opposed. Maybe others want to add their opinion?

Reply:
Yes, I meant asserting within the class (either only during debug mode or always)

Reply:
I added an assertion that the versions are incrementing: [`f0258ec` (#3380)](https://github.com/loculus-project/loculus/pull/3380/commits/f0258ec899abbd088dfb0218e2ccf1cd861fc84f)

I think with this and the test pointing to this class, it should be sufficient.

The assertion doesn't check whether all accessions are coming in a block together, but to keep track of that we'd need to track all accessions any ways, at that point we could even ditch the requirement for accessions coming in as chunks 😄 

IMO this is sufficient, so I'm marking this as resolved (also because you weren't sure whether to add anyways). If you think more should be done, feel free to un-resolve!

Reply:
sounds good! thanks!